node/src/components/chain_synchronizer/operations.rs

```diff
@@ -157,6 +157,16 @@ where
             },
         };
 
+        if trusted_block_header.protocol_version() != config.protocol_version()
+            && !config.is_last_block_before_activation(&trusted_block_header)
```

Comment:
(Once #3212 is done we can remove this check as well, right? Because if we start with the last block before activation, the initializer reactor will already have created the immediate switch block?)

Reply:
I think so :+1: